src/shared/sol-util.c

```diff
@@ -312,14 +315,37 @@ sol_util_str_split(const struct sol_str_slice slice, const char *delim, size_t m
 static int
 get_progname(char *out, size_t size)
 {
-    char cwd[PATH_MAX] = { NULL }, *execfn;
+    char cwd[PATH_MAX] = { NULL }, *execfn = NULL;
 
     if (getcwd(cwd, sizeof(cwd)) == NULL)
-        return -1;
+        return -errno;
 
+#ifdef HAVE_SYS_AUXV_H
     execfn = (char *)getauxval(AT_EXECFN);
```

Comment:
And old Linuxes. I thought of embedding an additional check (&& is_linux) there, but @dorileo commented that the using these functions does not even make sense on riot et al (single, static, binary).